pkg/limayaml/limayaml.go

```diff
@@ -229,6 +230,12 @@ type Provision struct {
 	Script                          string             `yaml:"script" json:"script"`
 	File                            *LocatorWithDigest `yaml:"file,omitempty" json:"file,omitempty" jsonschema:"nullable"`
 	Playbook                        string             `yaml:"playbook,omitempty" json:"playbook,omitempty"`
+	// The following fields must be nil unless Mode is ProvisionModeData
+	Content     *string `yaml:"content,omitempty" json:"content,omitempty" jsonschema:"nullable"`
+	Overwrite   *bool   `yaml:"overwrite,omitempty" json:"overwrite,omitempty" jsonschema:"nullable"`
+	Owner       *string `yaml:"owner,omitempty" json:"owner,omitempty"`
+	Path        *string `yaml:"path,omitempty" json:"path,omitempty"`
+	Permissions *string `yaml:"permissions,omitempty" json:"permissions,omitempty"`
```

Comment:
Shouldn't this be int?

Reply:
I had it as an `int` first, but that makes it hard/impossible to verify that the user specified a leading `0` to make it octal, so you wouldn't get a validation error for `permissions: 511`[^1]. The string is always parsed as octal, so it doesn't matter if you specify the leading `0` or not.

I thought this could be a common cause of errors because `chmod` doesn't require the leading `0` either.

[^1]: I just double-checked, and `511` is the same as `0777`, so you can't tell if the user meant `0511` or not.

Reply:
I just noticed that I didn't update the comments in `default.yaml`, which still claim that the leading `0` is required. That is a leftover from when the field was still an `int`.